src/SpectatorView.Unity/Assets/SpectatorView/Scripts/SpectatorView.cs

```diff
@@ -170,10 +195,17 @@ private void RunStateSynchronizationAsObserver()
         private void SetupRecordingService()
         {
 #if UNITY_ANDROID || UNITY_IOS
-            if (enableMobileRecordingService &&
-                mobileRecordingServiceVisualPrefab != null)
+            GameObject recordingVisualPrefab = defaultMobileRecordingServiceVisualPrefab;
+            if (MobileRecordingSettings.IsInitialized && MobileRecordingSettings.Instance.OverrideMobileRecordingServicePrefab != null)
+            {
+                recordingVisualPrefab = MobileRecordingSettings.Instance.OverrideMobileRecordingServicePrefab;
+            }
+
+            if (MobileRecordingSettings.IsInitialized && 
+                MobileRecordingSettings.Instance.EnableMobileRecordingService &&
+                recordingVisualPrefab != null)
             {
-                mobileRecordingServiceVisual = Instantiate(mobileRecordingServiceVisualPrefab);
+                mobileRecordingServiceVisual = Instantiate(recordingVisualPrefab);

```

Comment:
>mobileRecordingServiceVisual [](start = 16, length = 28)

Does this get cleaned up anywhere?

Reply:
I've added it to OnDestroy to be cleaned up there

---
In reply to: [298334939](https://github.com/microsoft/MixedReality-SpectatorView/pull/48#discussion_r298334939) [](ancestors = 298334939)